src/lib/services/setting-service.ts

```diff
@@ -44,20 +44,27 @@ export default class SettingService {
 
     async insert(id: string, value: object, createdBy: string): Promise<void> {
         const exists = await this.settingStore.exists(id);
+        let data = { id };
+        if (typeof value === 'object') {
+            data = { id, ...value };
+        } else {
+            this.logger.error('Invalid setting value, source: %s', id);
+        }
+
         if (exists) {
             await this.settingStore.updateRow(id, value);
             await this.eventService.storeEvent(
                 new SettingUpdatedEvent({
```

Comment:
I have sine doubts about this generic settings-updated events. Some things we store in the settings table should not be populated to the audit events, e.g. when we store SAML details, we want to avoid printing the private key in the audit logs.

Reply:
This has now been fixed, nice catch!